part3/solvers.py

```
@@ -0,0 +1,57 @@
+
+    for i in range(n):
+        if A[i][i] == 0:
+            raise ValueError(f"Phần tử trên đường chéo chính A[{i}][{i}] bằng 0, không thể chia được.")
+
+    # Tiêu chí 3: Kiểm tra chéo trội và cảnh báo
+    if not is_diagonally_dominant(A):
+        warnings.warn("Ma trận không chéo trội. Phương pháp Gauss-Seidel có thể không hội tụ.")
+
+    x = [0.0] * n  # Khởi tạo nghiệm ban đầu là vector 0
+    
+    for k in range(max_iter):
```

Ruff (0.15.9)

[warning] Loop control variable `k` not used within loop body

Rename unused `k` to `_k`

(B007)